### PR TITLE
Allow overriding rust-analyzer display version

### PR DESCRIPTION
**Pull request by oxalica**
The build script invokes `git` for version information which is displayed when rust-analyzer is called with `--version`. But in build environment without `git` or when the source code is not a git repo, there's no way to manually specify the version information.

This patch respects environment variable ~`REV`~ `RUST_ANALYZER_REV` in compile time for overriding.

Related: https://github.com/NixOS/nixpkgs/pull/90976

### CODE REVIEW

File: crates/rust-analyzer/build.rs [hunk 1]
```diff
@@ -5,11 +5,13 @@ use std::{env, path::PathBuf, process::Command};
 fn main() {
     set_rerun();
 
-    let rev = rev().unwrap_or_else(|| "???????".to_string());
+    let rev = env::var("REV").ok().or(rev()).unwrap_or_else(|| "???????".to_string());
```
Comment (bjorn3) [inline crates/rust-analyzer/build.rs:8]:
```suggestion
    let rev = env::var("RUST_ANALYZER_REV").ok().or_else(rev).unwrap_or_else(|| "???????".to_string());
```

Other crates may do something similar to get the revision name.

Comment (oxalica) [reply crates/rust-analyzer/build.rs:8]: Doesn't them share the same revision?

Comment (bjorn3) [reply crates/rust-analyzer/build.rs:8]: This is the rust analyzer revision, which uses either git commit hash or now a release date, other crates may use a different revision system. For example they may want to add a release channel (rustc) or something else.

File: crates/rust-analyzer/build.rs [hunk 2]
```diff
@@ -5,11 +5,13 @@ use std::{env, path::PathBuf, process::Command};
 fn main() {
     set_rerun();
 
-    let rev = rev().unwrap_or_else(|| "???????".to_string());
+    let rev = env::var("RUST_ANALYZER_REV").ok().or_else(rev).unwrap_or_else(|| "???????".to_string());
     println!("cargo:rustc-env=REV={}", rev)
 }
 
 fn set_rerun() {
+    println!("cargo:rerun-if-env-changed=REV");
```
Comment (bjorn3) [inline crates/rust-analyzer/build.rs:13]:
```suggestion
    println!("cargo:rerun-if-env-changed=RUST_ANALYZER_REV");
```

Comment (oxalica) [reply crates/rust-analyzer/build.rs]: Oh. Just forget to `git add`.

**Comment by kjeremy** [comment]
Does this need to change here too: https://github.com/rust-analyzer/rust-analyzer/blob/f4f51171ca6d99b693df2ef6fb71f0347999aa9f/crates/rust-analyzer/src/bin/main.rs#L93

**Comment by oxalica** [comment]
@kjeremy 
> Does this need to change here too:
> 
> https://github.com/rust-analyzer/rust-analyzer/blob/f4f51171ca6d99b693df2ef6fb71f0347999aa9f/crates/rust-analyzer/src/bin/main.rs#L93

Nope. The build script still prints `cargo:rustc-env=REV={}`, so `env!` will read from `REV` set by the build script.

**Pull request merged**